stan/math/prim/scal/fun/as_array_or_scalar.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_AS_ARRAY_OR_SCALAR_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_AS_ARRAY_OR_SCALAR_HPP
+
+namespace stan {
+namespace math {
+
+/**
+ * Converts a matrix type to an array. For scalar inputs this is identity
```

Comment:
This should just document this specialization.  So it can just be `Returns specified input value.`